modules/indexer/issues/db/db.go

```diff
@@ -51,26 +51,32 @@ func (i *Indexer) Search(ctx context.Context, options *internal.SearchOptions) (
 	//        The notification is defined in modules, but it's using lots of things should be in services.
 
 	cond := builder.NewCond()
-
 	if options.Keyword != "" {
 		repoCond := builder.In("repo_id", options.RepoIDs)
 		if len(options.RepoIDs) == 1 {
 			repoCond = builder.Eq{"repo_id": options.RepoIDs[0]}
 		}
-		subQuery := builder.Select("id").From("issue").Where(repoCond)
-
-		cond = builder.Or(
-			db.BuildCaseInsensitiveLike("issue.name", options.Keyword),
-			db.BuildCaseInsensitiveLike("issue.content", options.Keyword),
-			builder.In("issue.id", builder.Select("issue_id").
-				From("comment").
-				Where(builder.And(
-					builder.Eq{"type": issue_model.CommentTypeComment},
-					builder.In("issue_id", subQuery),
-					db.BuildCaseInsensitiveLike("content", options.Keyword),
-				)),
-			),
-		)
+
+		if options.Index.Has() {
+			cond = builder.And(
+				builder.Eq{"\"index\"": options.Keyword},
```

Comment:
I guess it could be this

```suggestion
				builder.Eq{"`index`": options.Keyword},
```

(not tested .....)

Reply:
I am sure that the double quotes will work with both sqlite and postgresql, not so sure about the others...

Will look into this.